### ISSUE TITLE
[FEATURE] Run integration tests as part of the OpenSearch distribution

### ISSUE THREAD

**Issue by peterzhuamazon** [opensearch-project/geospatial#115]
**Is your feature request related to a problem?**
Following a problem in another plugin (https://github.com/opensearch-project/opensearch-build/issues/2043), and coming from https://github.com/opensearch-project/opensearch-build/issues/58, there's no automated testing that this plugin runs as part of the OpenSearch distribution.

**What solution would you like?**
Run integration tests as part of the distribution.

**Comment by VijayanB** [comment]
@peterzhuamazon Can this closed?

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon Can this be closed?

Let us wait until the 2.2.0 release to make sure the tests are running well.

**Comment by peterzhuamazon** [comment]
Update: requires addition of security enabled tests before closing this issue.

**Comment by VijayanB** [comment]
@peterzhuamazon Did we try running integTestRemote with following properties
```
    systemProperty "https", System.getProperty("https")
    systemProperty "user", System.getProperty("user")
    systemProperty "password", System.getProperty("password")
```

**Comment by navneet1v** [comment]
> Update: requires addition of security enabled tests before closing this issue.

Can you please provide more details around what is the expectation around this?

**Comment by heemin32** [comment]
GeoSpatial remote integration test fails with security plugin enabled cluster. I see ssl connection error while running tests.

**Comment by peterzhuamazon** [comment]
Test on 6039 build of opensearch 2.3.0, using geospatial 2.x f66bb9a0c711073802ddd1d71670210bf004dc2b commit with these logs:

```
2> REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.geospatial.index.query.xypoint.XYPointQueryIT.testIndexPointsMultiPolygon" -Dtests.seed=5C68C869CC61223 -Dtests.security.manager=false -Dtests.locale=sq-AL -Dtests.timezone=Canada/Mountain -Druntime.java=11
  2> org.opensearch.client.ResponseException: method [PUT], host [http://localhost:9200/], URI [/dsyfegyoyi], status line [HTTP/1.1 400 Bad Request]
    {"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"No handler for type [xy_point] declared on field [upeagbfcmuoztzvv]"}],"type":"mapper_parsing_exception","reason":"Failed to parse mapping [_doc]: No handler for type [xy_point] declared on field [upeagbfcmuoztzvv]","caused_by":{"type":"mapper_parsing_exception","reason":"No handler for type [xy_point] declared on field [upeagbfcmuoztzvv]"}},"status":400}
```

Per @heemin32 this is a feature of xypoint that is not available in 2.3.0, but for 2.4.0.

We will come back and check again once 2.4.0 is on the release phase.

@heemin32 will update the test manifest adding `- with-security` once 2.4.0 test manifest is up.

Thanks.
